contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py

```diff
@@ -12,31 +12,89 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import threading
+import time
 
 from opencensus.ext.azure.common import Options
+from opencensus.metrics.export import value
+from opencensus.metrics.export.metric_descriptor import MetricDescriptorType
 from opencensus.metrics import transport
 from opencensus.stats import stats
+from opencensus.ext.azure.common import utils
+from opencensus.ext.azure.common.protocol import Data
+from opencensus.ext.azure.common.protocol import DataPoint
+from opencensus.ext.azure.common.protocol import Envelope
+from opencensus.ext.azure.common.protocol import MetricData
+from opencensus.ext.azure.common.transport import TransportMixin
 
 __all__ = ['MetricsExporter', 'new_metrics_exporter']
 
 
-class MetricsExporter(object):
+class MetricsExporter(TransportMixin):
     """Metrics exporter for Microsoft Azure Monitor."""
 
-    def __init__(self, options):
+    def __init__(self, options=None):
+        if options is None:
+            options = Options()
         self.options = options
         if not self.options.instrumentation_key:
             raise ValueError('The instrumentation_key is not provided.')
-        self._md_cache = {}
-        self._md_lock = threading.Lock()
 
     def export_metrics(self, metrics):
-        metrics = list(metrics)
+        if not metrics:
+            return
+        envelopes = []
         for metric in metrics:
-            # TODO: implement export to Azure Monitor functionality
-            print(repr(metric))
+            # No support for histogram aggregations
+            if metric.descriptor.type == MetricDescriptorType.CUMULATIVE_DISTRIBUTION:
+                continue
+            envelopes.append(self.metric_to_envelope(metric))
+            self._transmit(envelopes)
```

Comment:
Should this be inside the for loop or outside?

Reply:
Outside. Fixed.